e2e/cypress/integration/enterprise/saml/saml_automated_spec.js

```diff
@@ -0,0 +1,219 @@
+                                cy.findByTestId('auditTableBody').find('td').
+                                    each(($el) => {
+                                        cy.wrap($el).
+                                            invoke('text').
+                                            then((text) => {
+                                                if (text.includes('Saml obtained user')) {
+                                                    expect(text).to.contains('Saml obtained user');
+                                                }
+                                            });
+                                    });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        it('SAML Signature Algorithm using RSAwithSHA256', () => {
+            cy.apiAdminLogin();
+            const test1Settings = {
+                ...newConfig,
+                SamlSettings: {
+                    ...newConfig.SamlSettings,
+                    SignatureAlgorithm: 'RSAwithSHA256',
+                },
+            };
+            cy.apiUpdateConfig(test1Settings).then(() => {
+                testSettings.user = regular1;
+                cy.oktaGetOrCreateUser(testSettings.user).then((oktaUserId) => {
+                    cy.oktaDeleteSession(oktaUserId);
+                    cy.doSamlLogin(testSettings).then(() => {
+                        cy.doOktaLogin(testSettings.user).then(() => {
+                            cy.skipOrCreateTeam(testSettings, oktaUserId);
+                            cy.oktaDeleteSession(oktaUserId);
+                        });
+                    });
+                });
+            });
+        });
+
+        it('SAML Signature Algorithm using RSAwithSHA512', () => {
+            cy.apiAdminLogin();
+            const test1Settings = {
+                ...newConfig,
+                SamlSettings: {
+                    ...newConfig.SamlSettings,
+                    SignatureAlgorithm: 'RSAwithSHA512',
+                },
+            };
+            cy.apiUpdateConfig(test1Settings).then(() => {
+                testSettings.user = regular1;
+                cy.oktaGetOrCreateUser(testSettings.user).then((oktaUserId) => {
+                    cy.oktaDeleteSession(oktaUserId);
+                    cy.doSamlLogin(testSettings).then(() => {
+                        cy.doOktaLogin(testSettings.user).then(() => {
+                            cy.skipOrCreateTeam(testSettings, oktaUserId);
+                        });
+                    });
+                });
+            });
+        });
+
+        it('SAML2 Implementation', () => {
+            cy.apiAdminLogin();
+            const test1Settings = {
+                ...newConfig,
+                ExperimentalSettings: {
+                    UseNewSAMLLibrary: true,
+                },
+            };
```

Comment:
@saturninoabril, as per the expectations, this requires a server restart for the config to be effective. Any suggestions on how we can handle this while running our cypress test suite? My advise would be to use a Cypress env variable and run the tests with and without the NewSAML Library. This way we can have all the SAML related tests verified when this flag is turned on and off. We may not need a separate test for this. Please advise.

Reply:
@hahmadia can you please remove this testcase to use the newSAML Library. @saturninoabril and I had discussed this and we will now be running the SAML related cypress tests twice - i.e. with this config enabled and disabled. Since this requires a server restart for it to be effective, it needs to be controlled from outside and hence we don't need a test which changes it within the test (as it will not be effective). Thanks.

Reply:
Removed!